dotcom-rendering/src/lib/get-video-id.amp.ts

```diff
@@ -25,21 +24,10 @@ export const getIdFromUrl = (
 		.filter(isString)
 		.map((id) => id.slice(0, 11));
 
-	if (!ids.length)
-		logErr(
+	if (ids.length && ids[0]) return ids[0];
```

Comment:
Could we keep the logic as it was by returning early in an error?

```ts
const [id] = ids;
if(!id) throw Error('');
return id;
```

Reply:
Is there any code that would not be executed if we write it the other way around? I find this way a bit easier to read.

Reply:
The whole `ids.length` check is entirely unnecessary, as an undefined access will be `undefined` so we’re doing a bunch of unnecessary work. My proposal reduces indentation and the duplicate access to `ids[0]`. While we have run-to-completion guarantee in JavaScript, unifying your check for the existence of a value and using its value can reduce [time-of-check to time-of-use](https://en.wikipedia.org/wiki/Time-of-check_to_time-of-use) errors in general.

An bug-prone example may help demonstrate:

```ts
const map = new Map<string, number>();
/* … things get added to the map here … */
const hasThing = map.has('thing');
/* … thing gets removed from the map here, or maybe there’s an await … */
if(hasThing) {
  const thing = map.get('thing');
  console.info({ thing }); // logs `undefined`
}
```

```ts
const map = new Map<string, number>();
/* … things get added to the map here … */
const thing = map.get('thing');
/* … thing is either number or undefined … */
/* … any change to the map does not impact the value of thing! … */
if(thing !== undefined) {
  console.info({ thing }); // always logs a `number`
}
```

Reply:
I’m also surprised our linting rule allow the `logErr` to be on a following line without requiring curly braces – @arelra would get goosebumps.

The [relevant ESLint rule is `curly`](https://eslint.org/docs/latest/rules/curly#multi-or-nest)